### DISCUSSION TITLE
Motion Commander Tutorial Troubleshooting

### DISCUSSION THREAD

**Discussion by Itai51** [bitcraze/discussions#1417] [Q&A] [answered]
Hello Crazyflie community,

I am following the motion commander example from [this link](https://www.bitcraze.io/documentation/repository/crazyflie-lib-python/master/user-guides/sbs_motion_commander/). In the first section, "Step 1: Security before flying", I successfully run the code and get the following output:
`1
Deck is attached!`

However, when I try to run the example from the second section, "Step 2: Take off function", I get the following output (after the function times out):
`No flow deck detected!`

I've checked the wiring and tried reading through the code. I even tried deleting this condition, only for an erratic flight to take place.

What are possible reasons for the disparity?

Thank you!

**Comment by Itai51** [comment]
I forgot to mention that I also tried updating the CF2.1 firmware!

**Comment by knmcguire** [comment]
Hi! I saw that you also asked a question [here ](https://github.com/orgs/bitcraze/discussions/1391)with more details, but let's discuss further in this thread since that is better. Here is  a copy of my answer there:

> Hmmm.. this is strange! This would be a perfect surface for the Crazyflie's flowdeck to fly over. I feel that there is something else going on.
> 
> Could you perhaps test the crazyflie out by hand, and look at the plotting to see the variables motion.x, motion.y. Are they able to see any flow at all?
> 
> Also, do you see any error in the console tab as well?

It would be best to debug it further with the [cfclient](https://www.bitcraze.io/documentation/repository/crazyflie-clients-python/master/), where you can also make it take off in the flight tab

**Reply by Itai51** [reply, marked as answer]
I realized that when I copy each subsequent script at the end of each step, the previous functions are copied without any content. I copied the contents of param_deck_flow(...) from step 1 into step 2, and then it worked!
